app.rb

```diff
@@ -93,4 +93,41 @@ def list_rentals
     end
   end
 
+  def main_menu
+    puts ' '
+    puts 'Please choose an option by entering a number:
+        1 - List all books
+        2 - List all people
+        3 - Create a person
+        4 - Create a book
+        5 - Create a rental
+        6 - List all rentals for a given person id
+        7 - Exit'
+    puts ' '
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def menu_selection
+    main_menu
+    selection = gets.chomp.to_i
+    case selection
+    when 1
+      list_books
+    when 2
+      list_people
+    when 3
+      create_person
+    when 41
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals
+    when 7
+      puts 'Thanks for using our system. See you later'
+      exit
+    end
+    menu_selection
+    # rubocop:enable Metrics/CyclomaticComplexity
+  end
 end
```

Comment:
- [ ] Hi @GordonOtieno, I think you still need to refactor your App class. Based on the requirements of this project, requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose sole responsibilities are to request or validate input. Hence you'll still need to refactor the `create_person, create_book, create_rental, and list_rentals` methods. The methods `main_menu` and `menu_selection` can be placed in the main.js file since the main role of main.js is only to start the code and ask the user for an initial choice of options(or to quit the app). The "Single-responsibility Principle" should be thoroughly respected.